frontend/src/state/reducers/scheduleReducer.ts

```diff
@@ -120,7 +120,7 @@ export const scheduleReducer = (
         return draft;
       }
       case getType(setScheduleAction): {
-        const { schedule } = action.payload;
+        const schedule = JSON.parse(JSON.stringify(action.payload.schedule)); /// deep copy of schedule, because schedule is modified
```

Comment:
Why didn't we need a deep copy before, but do now?

Reply:
I mentioned this to you at the meeting, but it has something to do with React props being immutable, which caused an error.